server/webapp/dbmodels.py

```diff
@@ -660,32 +660,51 @@ def restore(self, progset, program_list):
                 program = self._program_to_dict(program)
                 update_or_create_program(self.project.id, self.id, program_name, program, True)
 
-    def create_programs_from_list(self, programs):
+    def recreate_programs_from_list(self, programs, progset_id):
         from optima.utils import saves
 
         prog_shorts = []
-        for program in programs:
-            kwargs = {}
-            for field in ['name', 'short', 'category', 'targetpops', 'pars', 'costcov', 'criteria']:
-                kwargs[field] = program[field]
+        desired_shorts = set([program.get('short', program.get('short_name', '')) for program in programs])
```

Comment:
should be the other way around, short_name is the default with fallback to short